packages/app/src/systems/CRX/background/services/BackgroundService.ts

```diff
@@ -341,4 +343,39 @@ export class BackgroundService {
     const abi = await AbiService.getAbi({ data: input.contractId });
     return abi;
   }
+
+  async addNetwork(
+    input: MessageInputs['addNetwork'],
+    serverParams: EventOrigin
+  ) {
+    const { network } = input;
+    if (!isValidNetworkUrl(network.url)) {
+      throw new Error('Invalid network URL');
+    }
+
+    const networkExists = await NetworkService.getNetworkFromUrl({
+      url: network.url,
+    });
+    if (networkExists) {
+      throw new Error('Network already exists');
+    }
```

Comment:
this network validation are getting too spread, we should create a function in NetworkService

`NetworkService.assertAddNetwork`

this method will have the necessary validations to add a network.

it should be reused in other places to clean